src/client/sse.ts

```diff
@@ -135,7 +130,9 @@ export class SSEClientTransport implements Transport {
             headers.set("Accept", "text/event-stream");
             const response = await fetchImpl(url, {
               ...init,
-              headers,
+              headers: {
+                ...Object.fromEntries(headers.entries()),
+              }
             })
```

Comment:
For the robustness, I also update the Object type of the `headers`, so it can be compatible with the spreadable declaration style (`...`) like before. 
Otherwise, if someone don't know the type change of `headers`, they still use the spreadable style to declare existing headers, like (`..init?.headers`), it will miss all existing headers without any sign, which may cause some tricky issues, e.g., in OAuth scenario.

e.g., the test case "refreshes expired token during SSE connection":
```TypeScript
eventSourceInit: {
  fetch: (url, init) => {
    return fetch(url, { ...init, headers: {
      ...init?.headers,
      'X-Custom-Header': 'custom-value'
    } });
  }
}
```